src/screens/subjects/hooks/useSubjects.ts

```diff
@@ -93,6 +93,12 @@ const useSubjects = () => {
       return;
     }
 
+    if (queryPage && !querySearch) {
```

Comment:
Como tem o return na linha 93 quando o `querySearch` é verdadeiro, ao chegar nessa linha ele sempre vai ser falso então não precisa fazer a checagem `!querySearch`: 

```suggestion
    if (queryPage) {
```